### ISSUE TITLE
Chat API notifications not being sent

### ISSUE THREAD

**Issue by Elipasousa** [zendesk/zendesk_sdk_chat_ios#181]
- SDK version: 1.3.7.1
- Xcode version: 9.2
- iOS or Simulator: both
- iOS Version: 10 and 11
- Swift version: 4.0

I'm implementing my own view controller using ZenDesk Chat APi notifications. I've noticed a couple of things:

1st) when an agent first joins the chat room, there's no `ZDCChatEventTypeMemberJoin` event notification being sent. The `agentEvents` is called with an array containing all the agents currently in the chat, but the `chatLogEvents` notification is not triggered. The only time I was able to receive the `ZDCChatEventTypeMemberJoin` (and `ZDCChatEventTypeMemberLeave`) notification, is when an agent leaves the chat room and then comes back. According to the documentation, member join and member leave are triggered when "member (agent) joined/left the chat". Not happening the first time.

2nd) when a rating request is sent from an agent, I receive a `ZDCChatEventTypeRating` that allows me to show my rating voting cell. After sending the user's vote, I allow for a rating comment to be made. So far, so good. The problem is when I send the rating comment, instead of receiving a `ZDCChatEventTypeRatingComment` notification with the rating comment, I get an update of the first `ZDCChatEventTypeRating` received, with a variable called `ratingComment` with the comment (that btw it's not on the documentation). `ZDCChatEventTypeRatingComment` is never triggered.

I based my implementation on the documentation found on [https://developer.zendesk.com/embeddables/docs/ios-chat-sdk/sessionapi](https://developer.zendesk.com/embeddables/docs/ios-chat-sdk/sessionapi)

**Comment by patrick-doyle** [comment]
Hi Elipasousa,

Can you provide a code sample of how the chatLogEvents and agentEvents observers are added to `ZDCChatAPI` instance to help with solving this issue?

**Comment by Elipasousa** [comment]
Hi Patrick, here's the methods I'm calling on my `viewDidLoad`. Please let me know if you need any extra info. Thanks

`        ZDCChatAPI.instance().addObserver(self, forConnectionEvents: #selector(onConnectionStateUpdated))
        ZDCChatAPI.instance().addObserver(self, forTimeoutEvents: #selector(timeoutEvent(_:)))
        ZDCChatAPI.instance().addObserver(self, forChatLogEvents: #selector(chatEvent(_:)))
        ZDCChatAPI.instance().addObserver(self, forAgentEvents: #selector(onAgentEvent))
        ZDCChatAPI.instance().addObserver(self, forUploadEvents: #selector(chatUploadEvent(_:)))
        ZDCChatAPI.instance().addObserver(self, forAccountEvents: #selector(accountEvent(_:)))
        
        let config = ZDCAPIConfig()
        config.department = "help"
        config.tags = ["ios_app"]
        ZDCChatAPI.instance().startChat(withAccountKey: "MY_KEY", config: config)`

**Comment by divo** [comment]
@Elipasousa Can you try listening for [Agent Events](https://developer.zendesk.com/embeddables/docs/ios-chat-sdk/sessionapi#agent-events) for notifications on Agents joining / leaving the chat? A chat session has not really started until an agent is active, hence why no initial chat event is sent for them joining. I know it's a bit unintuitive.

For No.2, `ZDCChatEventTypeRatingComment` never being fired is a bug on our end. We are aware of it but the information is available in the updated `ZDCChatEventTypeRating` so it has a pretty low priority for a fix. Thank you for taking the time to report it, they more people that do the more of a priority it becomes.

**Comment by Elipasousa** [comment]
Thanks @divo for taking the time to answer me. I found a workaround for both issues:

- regarding the first issue, I'm forcing an `ZDCChatEventTypeMemberJoin` for the first time the first agent starts typing.

- for the rating comment bug, I'm just checking for an update on `ratingComment ` from the original `ZDCChatEventTypeRating` event. The documentation could have been updated to include this variable.
